### PR TITLE
Add yodl Formula

### PR DESCRIPTION
**Pull request by zchee**
Possible build, but dirty.

- Will be builded `yodlverbinsert.dSYM` binary
- Check `latex` https://github.com/Homebrew/homebrew/pull/45546/files#diff-552ecc49b5d5eb76f4e9d9b4e5d5c94aR28

Could you give me some advice?

### CODE REVIEW

File: Library/Formula/yodl.rb [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    inreplace 'icmake/install', /run\("cp " \+ g_install \+ BIN/, 'run("cp -R " + g_install + BIN'
+
+    system "./build", "programs"
+    system "./build", "macros"
+    system "./build", "man"
+
+    system "./build", "install", "programs"
+    system "./build", "install", "macros"
+    system "./build", "install", "man"
+    unless `/usr/bin/which latex` then
```
Comment (DomT4) [inline Library/Formula/yodl.rb]: You should use `depends_on :tex => :optional` and then:

```ruby
if build.without? "tex"
  system "./build", "manual"
  system "./build", "install", "manual"
end
```

File: Library/Formula/yodl.rb [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  depends_on "ghostscript"
+  depends_on "icmake"
+
+  patch :DATA
+
+  depends_on "gnu-sed" => :build
+  
+  def install
+    inreplace 'INSTALL.im', /"\/usr"/, "\"#{prefix}\""
+    inreplace 'build', /\/usr\/bin\/icmake/, "/usr\/local\/bin\/icmake"
```
Comment (DomT4) [inline Library/Formula/yodl.rb]: You can't rely on `icmake` being in `/usr/local`.

File: Library/Formula/yodl.rb [hunk 3]
```diff
@@ -0,0 +1,50 @@
+require 'formula'
```
Comment (DomT4) [inline Library/Formula/yodl.rb]: Don't need this line.

File: Library/Formula/yodl.rb [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+class Yodl < Formula
+  url 'http://downloads.sourceforge.net/project/yodl/yodl/3.05.01/yodl_3.05.01.orig.tar.gz'
```
Comment (DomT4) [inline Library/Formula/yodl.rb]: Please use `https` on SF URLs.

File: Library/Formula/yodl.rb [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+  homepage 'http://yodl.sourceforge.net/'
```
Comment (DomT4) [inline Library/Formula/yodl.rb]: `homepage` should go above the `url`.

File: Library/Formula/yodl.rb [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+  sha1 '94d8e59a8569a9d6fb4dc7a1b8e7d430f018e570'
```
Comment (DomT4) [inline Library/Formula/yodl.rb]: This should be a `sha256`, and double quotes everywhere please.

File: Library/Formula/yodl.rb [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+
+  depends_on "ghostscript"
+  depends_on "icmake"
+
+  patch :DATA
```
Comment (DomT4) [inline Library/Formula/yodl.rb]: Please add an explanation of this patch & link to the upstream bug/submission, Thanks.

**Comment by zchee** [comment]
@DomT4 Fixed.
and, not required inline patch.

Correct define `icmake` binary?
https://github.com/Homebrew/homebrew/pull/45546/files#diff-552ecc49b5d5eb76f4e9d9b4e5d5c94aR14

===

Off the topic.
When I modify or create a new formula, command to test rather than the `install` or `reinstall` are implemented in brew?

**Comment by zchee** [comment]
@DomT4 ping.

If merged this PR, Possible `zsh` HEAD build!!

File: Library/Formula/yodl.rb [hunk 8]
```diff
@@ -0,0 +1,32 @@
+class Yodl < Formula
+  homepage 'http://yodl.sourceforge.net/'
```
Comment (DomT4) [inline Library/Formula/yodl.rb]: Still need double quotes around a lot of the lines here.

File: Library/Formula/yodl.rb [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+  url 'https://downloads.sourceforge.net/project/yodl/yodl/3.05.01/yodl_3.05.01.orig.tar.gz'
+  sha256 '5a3d0e1b2abbba87217cfdc6cd354a00df8d782572495bbddbdfbd4f47fe0d3e'
+
+  depends_on "ghostscript"
+  depends_on "icmake"
+
+  depends_on "gnu-sed" => :build
```
Comment (DomT4) [inline Library/Formula/yodl.rb]: Why GNU-sed? What breaks on the system one?

**Comment by DomT4** [comment]
```
== /usr/local/Library/Formula/yodl.rb ==
C:  2: 12: Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.
C:  3:  7: Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.
C:  4: 10: Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.
C: 13: 15: Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.
C: 13: 29: Use %r around regular expression.
C: 14: 15: Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.
C: 14: 24: Use %r around regular expression.
C: 16: 15: Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.
C: 32:  1: 1 trailing blank lines detected.

1 file inspected, 9 offenses detected
Error: 3 problems in 1 formula

==> audit problems
yodl:
 * A `test do` test block should be added
 * Formula should have a desc (Description).
 * Non-executables were installed to "/usr/local/Cellar/yodl/3.05.01/bin"
The offending files are:
  /usr/local/Cellar/yodl/3.05.01/bin/yodlverbinsert.dSYM
```

**Comment by zchee** [comment]
@DomT4 
Fixed.

but, I don't know how to solve this error.
```
yodl:
 * Non-executables were installed to "/usr/local/Cellar/yodl/3.05.01/bin"
```

**Comment by mikemcquaid** [comment]
@zchee You need to not install `.dSYM` files into `bin`. Probably you can just delete this.

**Comment by bfontaine** [comment]
Anything new here?

**Comment by bfontaine** [comment]
Closing due to no response. Feel free to open another PR addressing the comments above. Thanks!

**Pull request closed without merging**